gazebosim/ionic_demo
# Add bullet auto-deactivation demo

# 🎉 New feature

## Summary

Adds demo to showcase the auto-deactivation feature from bullet-featherstone

9 steak set models are added to the demo world. They are initially attached to the interior model using detachable joint, which makes them static. The joints can then be detached to make the models dynamic again. The RTF should drop for a few seconds until the models settle, and then the RTF should bounce back up.


![bullet-deativation-steak](https://github.com/user-attachments/assets/1b8a2a3f-7090-4bb7-aa8a-0cf7c4ca79dd)

## To Test

Launch the demo by running:

```
gz sim -v 4 -r ionic.sdf --physics-engine libgz-physics-bullet-featherstone-plugin
```

Wait a few seconds for RTF to settle.

Detach the joints:

```
gz topic -t "/steak/detach" -m gz.msgs.Empty -p "unused: true"
```

The RTF should drop. With max step size of 0.001, it's < 20% on my machine. Wait a few more and the RTF should bounce back up to 9X%.

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.
